wgpu-hal/src/metal/command.rs

```diff
@@ -653,9 +654,33 @@ impl crate::CommandEncoder for super::CommandEncoder {
                 descriptor
                     .set_visibility_result_buffer(Some(occlusion_query_set.raw_buffer.as_ref()))
             }
-
+            // This strangely isn't mentioned in https://developer.apple.com/documentation/metal/improving-rendering-performance-with-vertex-amplification.
+            // The docs for [`renderTargetArrayLength`](https://developer.apple.com/documentation/metal/mtlrenderpassdescriptor/rendertargetarraylength)
+            // also say "The number of active layers that all attachments must have for layered rendering," implying it is only for layered rendering.
+            // However, when I don't set this, I get undefined behavior in nonzero layers, so I think it is required.
```

Comment:
when you say you get undefined behavior you mean that you get crashes or that you get garbage value? I mean udef is hard  to "observe" as, well, the behavior would be unknown